src/main.rs

```diff
@@ -22,26 +22,37 @@ struct CmdArgs {
     workdir: PathBuf,
     #[structopt(short, long)]
     task_package: PathBuf,
+    #[structopt(long)]
+    debug: bool,
     #[structopt(subcommand)]
     command: Commands,
 }
 
 fn main() -> Result<()> {
-    let our_rust_log = "cranelift_wasm=warn,cranelift_codegen=info,wasi_common=info";
-    match env::var("RUST_LOG") {
-        Err(_) => env::set_var("RUST_LOG", our_rust_log),
-        Ok(var) => env::set_var("RUST_LOG", format!("{},{}", var, our_rust_log)),
-    };
-    env_logger::init();
-
     let cmdline = CmdArgs::from_args();
+
+    env_logger::from_env("YA_WASI_LOG")
+        .filter(Some("cranelift_codegen"), log::LevelFilter::Error)
+        .filter(Some("cranelift_wasm"), log::LevelFilter::Error)
+        .filter(
+            Some("wasi_common"),
+            if cmdline.debug {
+                log::LevelFilter::Info
```

Comment:
I think `Trace` would be a better fit here. It wouldn't explode in size while providing WASI syscall trace for easier debugging.